src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Windows.cs

```diff
@@ -104,13 +104,18 @@ void SetCurrent(
 					newPage.Toolbar ??= new Toolbar(newPage);
 					_ = newPage.Toolbar.ToPlatform(modalContext);
 
-					var windowManager = modalContext.GetNavigationRootManager();
-
-					if (windowManager.RootView is WindowRootView wrv)
+					// Hide titlebar on previous page
+					var previousContext = previousPage.FindMauiContext();
```

Comment:
Could include a device test? https://github.com/dotnet/maui/tree/main/src/Controls/tests/DeviceTests

Reply:
Sure -- lemme add that to the existing modal push/pop Windows tests